// Protocol Buffers for Firebird Base testing
// Copyright 2020 Firebird Project.  All rights reserved.
// https://www.firebirdsql.org/
// https://firebird-base.rtfd.io
//
// License: The MIT License
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"),
// to deal in the Software without restriction, including without limitation
// the rights to use, copy, modify, merge, publish, distribute, sublicense,
// and/or sell copies of the Software, and to permit persons to whom
// the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included
// in all copies or substantial portions of the Software.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

syntax = "proto3";

package firebird.base;

import "google/protobuf/any.proto";
import "google/protobuf/struct.proto";

// Test enumeration

enum TestEnum {
  option allow_alias = true ;

  TEST_UNKNOWN    = 0 ;
  TEST_READY      = 1 ;
  TEST_RUNNING    = 2 ;
  TEST_WAITING    = 3 ;
  TEST_SUSPENDED  = 4 ;
  TEST_FINISHED   = 5 ;
  TEST_ABORTED    = 6 ;

  // Aliases

  TEST_CREATED    = 1 ;
  TEST_BLOCKED    = 3 ;
  TEST_STOPPED    = 4 ;
  TEST_TERMINATED = 6 ;
}

// TestState message

message TestState {
  string   name = 1 ;
  TestEnum test = 2 ;
}

// TestCollection message

message TestCollection {
  string                       name       = 1 ;
  repeated TestState           tests      = 2 ;
  google.protobuf.Struct       context    = 3 ;
  google.protobuf.Struct       annotation = 4 ;
  repeated google.protobuf.Any supplement = 5 ;
}


